### ISSUE TITLE
Jul18 manual release tests: debian9

### ISSUE THREAD

**Issue by reszelaz** [sardana-org/sardana#900]
This issue is created for reporting the results of the manual tests for Jul18 release on debian-stretch.
Tested on Debian GNU/Linux 9 (stretch).

### Installation
- [x] Install Sardana from the tar.gz : `pip install <tarball_artifact_URL>`

### Create testing environment and run testsuite
- [x] Start Pool demo2. In a console do `Pool demo2`.
- [x] Start MacroServer demo2 and connect to the Pool demo2.
  In another console do: `MacroServer demo2`
- [x] Set MacroServer's MacroPath to point to the macro examples.
  In another IPython console do:
  `PyTango.DeviceProxy('macroserver/demo2/1').put_property({'MacroPath':'<path_to_sardana_installation_dir>/macroserver/macros/examples'})`
- [x] Restart MacroServer e.g. Ctrl+C in the MacroServer's console and
  start it again.
- [x] Create spock profile demo2. In another console do `spock --profile=demo2`
- [x] In spock run `sar_demo` macro.
- [x] Edit `<path_to_sardana_installation_dir>/sardanacustomsettings.py`
  to point to the demo2 door e.g. `UNITTEST_DOOR_NAME = "door/demo2/1"`
- [x] Run testsuite. In another console do `sardanatestsuite`
  **Note**:
  ```
  ----------------------------------------------------------------------
  Ran 184 tests in 661.069s
  OK (skipped=7, expected failures=1)
  ```

### Test Sardana using Spock
- [x] Test interactive macros from spock e.g. `ask_for_moveable`, `ask_peak`
- [x] Execute `umvr` macro and verify that the position updates arrives.
- [x] Test `expconf`:
  1. Configure scan files using expconf set ScanDir to: `/tmp/` and
     ScanFile to: `demo1.h5, demo1.dat`.
  2. Configure online plot to show counters: On expconf GUI select for all the counter channels, Plot Type 'Spectrum' and Plot Axes '<mov>'
  3. Configure snapshot group: with a motor and the `sys/tg_test/1/ampli`
     attribute.
  4. Add the `sys/tg_test/1/double_scalar` attribute to the measurement
     group.
  5. Open online plot.
  6. Set JsonRecorder to true. In spock do `senv JsonRecorder True`
  7. Run step scan and verify if:
     **Warning:** After previously aborting `umvr` macro and then running the scan the outputs of the `umvr` were still appearing mixed with the scan output, see [output](https://gist.github.com/reszelaz/e13203accad25dd3bb1670b5f37641ce).
     - Records appear in spock output.
     - Records were stored in scan files.
     - Records were plotted on the online plot
- [ ] Run `showscan` and access to the last scan data.
  **Error**: #910
- [x] Test `edmac`:
  1. Modify existing macro: `ask_peak` and run it to verify that the change
     was applied.
  2. Create new macro in a new macro library:
     `edmac my_macro <path_to_sardana_installation_dir>/macroserver/macros/examples/my_macros.py
     and run it.

### Test Sardana with TaurusGUI

- [x] Create the GUI using this [guide](https://sourceforge.net/p/sardana/wiki/Howto-GUI_creation)

#### PMTV (PoolMotorTaurusValue)
- [x] Move motors from the slit panel in absolute and relative modes.
- [x] Show expert view.
- [x] Show compact mode.

#### macroexecutor
- [x] Execute `ascan` macro
- [x] Pause it in the middel and resume
- [x] Abort it
- [x] Add it to favorites
- [x] Run `lsm` macro
- [x] Execute `ascan` from favorites
- [x] Run `lsmac` macro
- [x] Execute `ascan` from history
- [x] Edit `dscan` macro in spock yellow line and run it
- [x] Restart macroexecutor application
- [x] Run `lsm` from history
- [x] Run `ascan` from favorites

#### sequencer
- [x] Add `ascan` macro to the sequence
- [x] Add `lsct` macro as a `post-acq` hook of `ascan`
- [x] Add `dscan` macro to the sequence
- [x] Run the sequence
- [x] Save sequence to a file
- [x] Start new sequence
- [x] Load sequence from a files
- [x] Run the loaded sequence

#### sardanaeditor
- [x] Open sardanaeditor with macroserver name as argument.
- [x] Browse macro libraries and open an existing macro.
- [x] Edit existing macro and save & apply chaneges.
  **Note:** see issue with "ctrl + s" #911 
- [x] Execute macro to see if changes were aplied.
- [x] Create a new macro using template.
- [x] Execute the newly created macro.

**Comment by teresanunez** [comment]
Hi @reszelaz, I have tested it in debian9. I see some problems with the GUIs, but I do not have the
last version of taurus installed:
-> I could not install it with pip (but  I think this is our problem, not a sardana one)
-> The +/- buttons in the motor gui do not make anything
-> Trying to create a new macro with the sardanaeditor gives only as possible selected libraries the .pyc files in the macroserver/macros directory, even if I try to select one to see what happens, I get an
error message telling that that directory is not in the macropath (it is not, but it does not have to be).

**Comment by daneos** [comment]
Hi @teresanunez,

> I could not install it with pip (but I think this is our problem, not a sardana one)

I also have some problems with pip on Debian 9, I couldn't even install PyTango.

> The +/- buttons in the motor gui do not make anything

Aren't the +/- buttons only an indication for limit switches?

**Comment by teresanunez** [comment]
> Hi @teresanunez,
> 
>     I could not install it with pip (but I think this is our problem, not a sardana one)
> 
> I also have some problems with pip on Debian 9, I couldn't even install PyTango.
>

Ok, I also have the problem with taurus.
 
>     The +/- buttons in the motor gui do not make anything
> 
> Aren't the +/- buttons only an indication for limit switches?
Yes, you are right !

**Comment by reszelaz** [comment]
Thanks @teresanunez for the tests!

> I have not tested the sequencer, since a discussion is going on about it.

There is just one minor issue with sequencer, present there for many releases, it is #305. The rest of the widget works perfectly.

> Trying to create a new macro with the sardanaeditor gives only as possible selected libraries the .pyc files in the macroserver/macros directory, even if I try to select one to see what happens, I get an
error message telling that that directory is not in the macropath (it is not, but it does not have to be).

I was able to create a new macro function using the sardanaeditor. I consider the sardanaeditor results satisfactory, I mean comparable with the previous releases. Anyway could you please report the issue of pyc it was always like this, but it is a bug. The same for the thing with macropath with the exact steps on how to reproduce this.

I have repeated the tests using taurus 4.4.0 cause there was a quite big risk due to [taurus-org/taurus#738](https://github.com/taurus-org/taurus/pull/738).

The is one release critical issue - see #912. Whenever this issue gets fixed I see debian 9 tests as passed!

**Comment by teresanunez** [comment]
Hi @reszelaz, I confirm the problem described in #911.

**Comment by teresanunez** [comment]
Hi @reszelaz, I found the problem with the sardanaeditor. I used capital letters in the name of the directory where I put the new sardana code, but in the msmacromanager the path is put in lower
case, so then it is not in the macropath, since the directories in the macropath keep the capital.
If I put my sardana installation in a directory with only lower case letter then I do not have any problem
 (well one see the .pyc files for selecting, but one select one and then everything works fine, the selected file is the .py one). So there is a missmatch between upper and lower cases affecting the
 sardanaeditor.
